### PR TITLE
Feature/hounslow ch440 add eligibility to search

### PR DESCRIPTION
**Pull request by fred-aghq**
### Summary

https://app.clubhouse.io/ayup-digital-ltd/story/440/add-eligibility-to-search

Write a brief short summary about the PR here.

### Development checklist

- [x] Changes have been made to the API?
  - [x] If so, the OpenAPI specification has been updated
- [ ] Database migrations have been added?
  - [ ] If so, the MySQL Workbench ERD has been updated
- [x] The code has been linted `./develop composer fix:style`

### CODE REVIEW

File: app/Docs/Schemas/Search/StoreSearchSchema.php [hunk 1]
```diff
@@ -46,7 +46,11 @@ public static function create(string $objectId = null): BaseObject
                             ->type(Schema::FORMAT_FLOAT)
                     ),
                 Schema::integer('distance')
-                    ->default(config('ck.search_distance'))
+                    ->default(config('ck.search_distance')),
+                Schema::array('service_elgibilities')
```
Comment (fred-aghq) [inline app/Docs/Schemas/Search/StoreSearchSchema.php:50]: typo!

File: app/Models/Service.php [hunk 2]
```diff
@@ -480,4 +484,31 @@ public static function placeholderLogo(int $maxDimension = null)
...
+        $searchableTaxonomies = $topLevelParentCategories->flatMap(function (Taxonomy $topLevelParent) use ($attachedParents) {
+            if (!in_array($topLevelParent->id, $attachedParents->toArray())) {
+                return $topLevelParent->children()->get()->map(function (Taxonomy $child) {
+                    return $child->name;
+                })->toArray();
+            }
+        });
+
+        return $searchableTaxonomies->merge($attachedEligibilityNames)->toArray();
+    }
```
Comment (appsol) [inline app/Models/Service.php:513]: Do we still need this?
If not better removed.

File: app/Search/ElasticsearchSearch.php [hunk 3]
```diff
@@ -279,6 +280,80 @@ public function applyRadius(Coordinate $location, int $radius): Search
...
+                                    ],
+                                ],
+                            ],
+                        ],
+                    ],
+                    'negative_boost' => 0.5,
+                ],
+            ],
+        ];
+
```
Comment (appsol) [inline app/Search/ElasticsearchSearch.php:341]: By writing over the whole query object, are you not preventing a combination query? e.g. Search for Age 12-15 + is free + within geo area?

File: tests/Feature/SearchTest.php [hunk 4]
```diff
@@ -602,4 +602,230 @@ public function test_services_with_more_taxonomies_in_a_persona_collection_are_m
...
+
+        // In this order
+        $this->assertEquals($serviceD->id, $data[0]['id']);
+        $this->assertEquals($serviceC->id, $data[1]['id']);
+        $this->assertEquals($serviceB->id, $data[2]['id']);
+        $this->assertEquals($serviceA->id, $data[3]['id']);
+
+        // And the inactive one is filtered out
+        $response->assertJsonMissing(['id' => $serviceIA->id]);
+    }
```
Comment (appsol) [inline tests/Feature/SearchTest.php:830]: Would be good to have 1 test that is a combo of multiple search terms, not worried about what it brings back (too hard to predict) so long as it's 200.

**Review by appsol — CHANGES_REQUESTED** [review]
See individual comments.

File: app/Models/Service.php (hunk 2 above)
Comment (fred-aghq) [reply app/Models/Service.php:513]: I agree, however at the moment it is still used by the ServiceEligibilityResource within ServiceResource.

I'll look at refactoring once I've addressed the more urgent issues.

**Review by appsol — APPROVED** [review]
Looks good.

**Pull request merged**